gateway/tests/e2e/providers/common.rs

```diff
@@ -741,6 +744,10 @@ pub async fn test_inference_params_inference_request_with_provider(provider: E2E
         "stream": false,
     });
```

Comment:
@ChetanXpro we will need to also extend the `tensorzero.toml` file to contain a model and provider for each provider type with `credential_location = dynamic::openai_api_key` or similar. Then we'll need to make a variant that uses this model for each of the provider types and use that for inference tests.  This can't work as-is I believe (the gateway won't know where to look).

Reply:
Note: I also prefer the lowercase snake case for the dynamic credentials as I show above since it's going to be the key in a JSON.